File: warmup/python2/maximizing-xor.py
#!/bin/python

# Complete the function below.


def maxXor(l,  r):
    xorList = []
    for i in range(l, r + 1):
        for j in range(l, r + 1):
            xorList.append(i ^ j)
    return max(xorList)

_l = int(raw_input())


_r = int(raw_input())

res = maxXor(_l, _r)
print(res)

File: warmup/c/maximizing_xor.c
#include <stdio.h>
/*
 * Complete the function below.
 */
int maxXor(int l, int r) {
    int maxVal = -1;
    int val;

    for(int i=l;i<=r;i++) {
        for(int j=l;j<=r;j++) {
            val = i ^ j;

            if (val > maxVal)
                maxVal = val;
        }
    }
    return maxVal;
}
int main() {
    int res;
    int _l;
    scanf("%d", &_l);

    int _r;
    scanf("%d", &_r);

    res = maxXor(_l, _r);
    printf("%d", res);

    return 0;
}
